CHANGELOG.md

```diff
@@ -20,6 +20,11 @@
 
 **Internal changes**:
 
+- Major reorganization of `prqlc-parser`. Files have been re-arranged and
+  exports have changed
+
+- Deprecated `prqlc-ast`. Functionality and objects were moved to `prqlc-parser`
```

Comment:
```suggestion
- Major reorganization of `prqlc-parser` — `prqlc-ast` is merged into `prqlc-parser`, and `prqlc-parser`'s files are rearranged, including its exports. This is part of an effort to modularize the compiler by stage, reducing the amount of context that's required to understand a single stage. There will likely be some further changes (more detail in the PR description). (@m-span, #4634)
  -  This is a breaking change for any libraries that use `prqlc-parser` which should be fairly rare.
```

(I actually wiped it `prqlc-ast`; don't think there's any need to keep it around...)